src/server/firefox/ffConnection.ts

```diff
@@ -111,12 +120,13 @@ export class FFConnection extends EventEmitter {
     this._closed = true;
     this._transport.onmessage = undefined;
     this._transport.onclose = undefined;
-    for (const callback of this._callbacks.values())
-      callback.reject(rewriteErrorMessage(callback.error, `Protocol error (${callback.method}): Target closed.`));
-    this._callbacks.clear();
+    const formattedBrowserLogs = helper.formatBrowserLogs(this._browserLogsCollector.recentLogs());
     for (const session of this._sessions.values())
-      session.dispose();
+      session.dispose(formattedBrowserLogs);
     this._sessions.clear();
+    for (const callback of this._callbacks.values())
+      callback.reject(rewriteErrorMessage(callback.error, `Protocol error (${callback.method}): Browser closed.` + formattedBrowserLogs));
```

Comment:
Maybe format the error once an pass it to callbacks/sessions?

Reply:
They use different `callback.method` unfortunately.